src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -215,6 +226,151 @@ def add_imaging_plane(nwbfile: NWBFile, metadata: dict, imaging_plane_index: int
+    nwbfile: NWBFile,
+    metadata: Optional[dict],
+    plane_segmentation_index: int = 0,
+    iterator_options: Optional[dict] = None,
+    compression_options: Optional[dict] = None,
+) -> NWBFile:
+    """
+    Adds the plane segmentation specified by the metadata to the image segmentation.
+    If the plane segmentation already exists in the image segmentation, it is not added again.
+
+    Parameters
+    ----------
+    segmentation_extractor : SegmentationExtractor
+        The segmentation extractor to get the results from.
+    nwbfile : NWBFile
+        The nwbfile to add the plane segmentation to.
+    metadata : dict, optional
+        The metadata for the plane segmentation.
+    plane_segmentation_index: int, optional
+        The index of the plane segmentation to add.
+    iterator_options : dict, optional
+        The options to use when iterating over the image masks of the segmentation extractor.
+    compression_options : dict, optional
+        The options to use when compressing the image masks of the segmentation extractor.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the plane segmentation added.
+    """
+    if iterator_options is None:
+        iterator_options = dict()
+    if compression_options is None:
+        compression_options = dict()
+
+    def image_mask_iterator():
+        for roi_id in segmentation_extractor.get_roi_ids():
+            image_masks = segmentation_extractor.get_roi_image_masks(roi_ids=[roi_id]).T.squeeze()
+            yield image_masks
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    plane_segmentation_metadata = image_segmentation_metadata["plane_segmentations"][plane_segmentation_index]
+    plane_segmentation_name = plane_segmentation_metadata["name"]
+
+    add_imaging_plane(
+        nwbfile=nwbfile,
+        metadata=metadata_copy,
+        imaging_plane_index=plane_segmentation_index,
+    )
+
+    add_image_segmentation(
+        nwbfile=nwbfile,
+        metadata=metadata_copy,
+    )
+
+    ophys = get_module(nwbfile, "ophys")
+    image_segmentation_name = image_segmentation_metadata["name"]
+    image_segmentation = ophys.get_data_interface(image_segmentation_name)
+
+    # Check if the plane segmentation already exists in the image segmentation
+    if plane_segmentation_name not in image_segmentation.plane_segmentations:
+        roi_ids = segmentation_extractor.get_roi_ids()
+        accepted_ids = [int(roi_id in segmentation_extractor.get_accepted_list()) for roi_id in roi_ids]
+        rejected_ids = [int(roi_id in segmentation_extractor.get_rejected_list()) for roi_id in roi_ids]
+
+        roi_locations = np.array(segmentation_extractor.get_roi_locations()).T
```

Comment:
Is the transpose here so the roi locations are written as column, row?

Reply:
yes; as far as I understood, `get_roi_locations()` returns roi locations as
row (x,y), column (num_rois) and to write it in the ROI table we need to have in column, row.

Reply:
Got you. I will correct the docstrings in `get_roi_locations` to indicate this more clearly. I did not find them very clear over there.